tfjs-backend-webgpu/src/shader_preprocessor.ts

```diff
@@ -79,26 +79,29 @@ export function makeShader(
   `);
 
   let uniformDeclaration = '';
-  program.variableNames.forEach((x, i) => {
-    uniformDeclaration += `${getCoordsDataType(inputInfo[i].shape.length)} ${
-        x.charAt(0).toLowerCase() + x.slice(1)}Shape; `;
-    prefixSnippets.push(`
-      layout(std430, set = 0, binding = ${1 + i}) readonly buffer ssb${x} {
-        ${mapToGlslTypes(inputInfo[i].dtype)} ${x}[];
-      };
-    `);
-  });
+  const variableNamesIsArray = Array.isArray(program.variableNames);
```

Comment:
I think as long as you initialize variableNames in the FillProgram to [], you shouldn't need this boolean.